apps/ewallet_db/lib/ewallet_db/blockchain_wallet.ex

```diff
@@ -0,0 +1,85 @@
+# Copyright 2018-2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule EWalletDB.BlockchainWallet do
+  @moduledoc """
+  Ecto Schema representing a blockchain wallet.
+  """
+  use Ecto.Schema
+  use ActivityLogger.ActivityLogging
+  import Ecto.Changeset
+  import EWalletDB.Validator
+  import EWalletDB.Helpers.Preloader
+
+  alias Ecto.UUID
+  alias EWalletDB.{Repo, BlockchainWallet}
+
+  @primary_key {:uuid, UUID, autogenerate: true}
+  @timestamps_opts [type: :naive_datetime_usec]
+
+  schema "blockchain_wallet" do
+    # Blockchain wallets don't have an external ID. Use `address` instead.
+    field(:address, :string)
+    field(:name, :string)
+    field(:public_key, :string)
+    field(:type, :string)
+    activity_logging()
+    timestamps()
+  end
+
+  defp changeset(%BlockchainWallet{} = wallet, attrs) do
+    wallet
+    |> cast_and_validate_required_for_activity_log(
+      attrs,
+      cast: [
+        :uuid,
+        :address,
+        :public_key,
+        :name,
+        :type
+      ],
+      required: [:address, :name, :public_key, :type]
```

Comment:
Should we require `:name` if this `BlockchainWallet` will also be user's wallets in the future?

Reply:
So, after discussing with Tibo, we should keep it required for now and potentially remove it later if needed.